frontend/src/app/selectors/experiment.js

```diff
@@ -25,6 +25,31 @@ export const launchedExperimentSelector = createSelector(
 );
 
 
-export default store => (
-  store.browser.isDev ? allExperimentSelector(store) : launchedExperimentSelector(store)
+// If not in a dev environment, filters to only include launched experiments.
+export const launchedExperimentSelector = store => (
+  store.browser.isDev ? allExperimentSelector(store) :
+                        onlyLaunchedExperimentSelector(store)
+);
+
+
+// Return the user's primary language subtag.
+export const localeSelector = store => store.browser.locale;
+
+
+// Passed a locale and set of experiments, filters out the experiments that are
+// blockedlisted in that locale, or grantlisted and not available in that locale.
+export const l10nSelector = (locale, experiments) => experiments.filter(exp => {
+  if ('locale_blocklist' in exp) {
+    return exp.locale_blocklist.indexOf(locale) === -1;
```

Comment:
Not sure if you want to use the fancy `Array#includes()` here instead. Doesn't really add anything, but you may be able to reduce to:

```js
return !exp.locale_blocklist.includes(locale);
...
return exp.locale_grantlist.includes(locale);
```